### PR TITLE
copy test logging stuff from IntegrationTesting here

### PR DESCRIPTION
**Pull request by anurse**
@davidfowl really wanted this available as low as possible, so I've moved it from IntegrationTesting to here, Logging.Testing. This is a noship package already.

@Eilon The Serilog dependency is moving once again, from Hosting to here. Once this is merged, I'll be converting the IntegrationTesting package to use it and removing the dependency there.

I also restructured this to allow me to write some tests for it.

### CODE REVIEW

File: src/Microsoft.Extensions.Logging.Testing/AssemblyTestLog.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
+﻿using System;
```
Comment (natemcmaster) [inline src/Microsoft.Extensions.Logging.Testing/AssemblyTestLog.cs]: Nit: copyright header

File: src/Microsoft.Extensions.Logging.Testing/LoggedTest.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
```
Comment (natemcmaster) [inline src/Microsoft.Extensions.Logging.Testing/LoggedTest.cs]: Copyright

File: test/Microsoft.Extensions.Logging.Testing.Tests/AssemblyTestLogTests.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
+﻿using System;
```
Comment (natemcmaster) [inline test/Microsoft.Extensions.Logging.Testing.Tests/AssemblyTestLogTests.cs]: Copyright

**Comment by anurse** [comment]
:up: :date:

**Pull request merged**